cdn_lambda/functions/origin_response/origin_response.py

```diff
@@ -0,0 +1,29 @@
+import logging
+import re
+
+LOG = logging.getLogger("origin-response")
+
+
+def lambda_handler(event, context):
+    # pylint: disable=unused-argument
+
+    request = event["Records"][0]["cf"]["request"]
+    original_uri = request["headers"]["original-uri"][0]["value"]
+
+    response = event["Records"][0]["cf"]["response"]
+
+    max_age_pattern_whitelist = [
+        ".+/PULP_MANIFEST",
+        ".+/listing",
+        ".+/repodata/repomd.xml",
+        ".+/ostree/repo/refs/heads/.+/.+",
+    ]
+
+    for pattern in max_age_pattern_whitelist:
+        if re.search(pattern, original_uri):
```

Comment:
It seems that either the patterns should end with $ or this should be an re.match, to ensure the pattern matches all the way up to the end.